### PR TITLE
Change Kubernetes membership to include GitHub org membership

### PR DESCRIPTION
**Pull request by pwittrock**
- Membership includes GH org membership
- Requirements and expectations increased due to increased privs
- More explanation of sponsorship

### CODE REVIEW

File: community-membership.md [hunk 1]
```diff
@@ -19,6 +19,9 @@ scoped by these sub-areas.
 [**New contributors**](https://github.com/kubernetes/contrib/issues/1090) should be welcomed to the community
 by existing members, helped with PR workflow, and directed to relevant documentation and communication channels.
 
+**Note:** Anyone can be added as an outside collaborator (with READ access) to a repo in the Kubernetes GitHub organization
```
Comment (bgrant0607) [inline community-membership.md]: There still needs to be a process for this, and some nominal requirements.

I believe only repo admins and org owners can add outside collaborators.

Comment (pwittrock) [inline community-membership.md]: @bgrant0607 FWIW I have been able to add folks as outside collaborators.

Comment (pwittrock) [inline community-membership.md]: Updated with some process and requirements for outside collaborators.

**Note:** I also update the sponsorship to require reviewers as sponsors to encourage more folks to work towards reviewer status.

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
@pwittrock @calebamiles @idvoretskyi @brendandburns @jbeda @sarahnovotny @philips @thockin @smarterclayton 

I'm going to merge this.

NOTEs:
* We need to set up the new mailing list, populate it with some org owners, and announce the new policy and process, so that the hordes of people who've requested membership in the past have the opportunity to request it.
* We probably will need to automate membership management, both for adding members and for monitoring activity levels necessary for continued membership. We've discussed sync'ing team memberships from files in a repo. Perhaps we could do the same for org memberships. (With some safeguards to not brick our org.)
* We should eliminate the org-level projects and convert them to repo projects.
* We also will need to monitor github changes more closely. Maybe contribX can regularly check the github blog or something?

**Comment by pwittrock** [comment]
> We need to set up the new mailing list, populate it with some org owners

New group created [here](https://groups.google.com/forum/#!managemembers/kubernetes-membership).  Sent out invitations to folks mentioned in your comment

> and announce the new policy and process, so that the hordes of people who've requested membership in the past have the opportunity to request it.

SGTM.  We could additionally announce at the community meeting today.

**Pull request merged**